sky/task.py

```diff
@@ -104,6 +105,36 @@ def replace_var(match):
     return json.loads(file_mounts_str)
 
 
+def _with_docker_login_config(
+    resources_set: Set['resources_lib.Resources'],
+    task_envs: Dict[str, str],
+) -> 'resources_lib.Resources':
+    all_keys = {
+        constants.DOCKER_USERNAME_ENV_KEY,
+        constants.DOCKER_PASSWORD_ENV_KEY,
+        constants.DOCKER_SERVER_ENV_KEY,
+    }
+    existing_keys = all_keys & set(task_envs.keys())
+    if not existing_keys:
+        return resources_set
+    if len(existing_keys) != len(all_keys):
+        with ux_utils.print_exception_no_traceback():
+            raise ValueError(
+                f'If any of {", ".join(all_keys)} is set, all of them must '
+                f'be set. Missing envs: {all_keys - existing_keys}')
+    docker_login_config = command_runner.DockerLoginConfig.from_env_vars(
+        task_envs)
+
+    def _add_docker_login_config(resources: 'resources_lib.Resources'):
+        if resources.extract_docker_image() is None:
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError(
+                    'Docker login config is only supported for docker images.')
```

Comment:
How about we just skip it instead of error out here? Reason: 
```suggestion
        if resources.extract_docker_image() is None:
            logger.warning('{colorama.Fore.YELLOW}Docker login configs {", ".join(all_keys)} are provided, but no docker image is specified in `image_id`. The login configs will be ignored.{colorama.Style.RESET_ALL}'
            return resources
```